src/containers/BooksForm.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import uniqid from 'uniqid'
+import { connect } from 'react-redux'
+import { createBook } from '../actions'
+
+const bookCategories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+]
+
+class BooksForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      title: '',
+      category: bookCategories[0],
+    }
+  }
+
+  handleChange = event => {
+    const { name, value } = event.target
+    this.setState({
+      [name]: value,
+    })
+  }
+
+  handleSubmit = event => {
+    event.preventDefault()
+    const { title, category } = this.state
+    const { createBook } = this.props
+    createBook({
+      id: uniqid(),
+      title,
+      category,
+    })
+    event.target.reset()
+  }
+
```

Comment:
After creating book, you need to RESET the state --> https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png

Reply:
Thanks! I miss that 😄 .